radicle-remote-helper/src/lib.rs

```diff
@@ -61,6 +65,16 @@ pub fn run(profile: radicle::Profile) -> Result<(), Box<dyn std::error::Error +
         .ok_or_else(|| Error::InvalidRemoteUrl(url.clone()))?
         .parse()
         .map_err(|e| Error::InvalidRemoteId(url.clone(), e))?;
+    let public_key: PublicKey = if url.path.is_empty() {
```

Comment:
Can I suggest we have a `Url` type in here to show what's going on a bit better and to have conversion traits? :)

Reply:
You want to re-implement `Url` just for the remote helper? Not sure I understand.

Reply:
I'm thinking something like:
```rust
pub struct Url {
  id: radicle::identity::Id,
  key: Option<PublicKey>,
}

impl TryFrom<&str> for Url {
  type Error = Error;
  fn try_from(s: &str) -> Result<Self, Self::Error> {
    radicle::git::Url::try_from(s).and_then(Self::try_from)
  }
}

impl TryFrom<radicle::git::Url> for Url {
  type Error = Error;
  fn try_from(s: &radicle::git::Url) -> Result<Self, Self::Error> {
    /* parsing logic here */
  }
}
```

Reply:
Ohh, gotcha, yes.

Reply:
Done.